packages/flutter/lib/src/services/text_input.dart

```diff
@@ -1058,6 +1078,24 @@ class TextInput {
     }
 
     final List<dynamic> args = methodCall.arguments as List<dynamic>;
+
+    if (method == 'TextInputClient.updateEditingStateWithTag') {
+      final TextInputClient client = _currentConnection._client;
+      assert(client != null);
+      if (client is AutofillTrigger) {
```

Comment:
shall we throw an error if the client is not AutofillTrigger?  updateEditingStateWithTag should only be called if there is autofill, right?

Reply:
If there's a client mismatch the message should be discarded (without throwing) I think, because it can happen if the user switches to a different input field too quickly after completing an autofill.